hid-uclogic-core.c

```diff
@@ -414,8 +414,27 @@ static int uclogic_raw_event_frame(
 
 	/* If need to, and can, transform the bitmap dial reports */
 	if (frame->bitmap_dial_byte > 0 && frame->bitmap_dial_byte < size) {
-		if (data[frame->bitmap_dial_byte] == 2) {
+		switch (data[frame->bitmap_dial_byte]) {
+		case 2:
 			data[frame->bitmap_dial_byte] = -1;
+			break;
+
+		/* Everything below here is for tablets that shove multiple dials into 1 byte */
+		case 4:
+			data[frame->bitmap_dial_byte] = 1;
```

Comment:
Mad brain fart moment. I should just squash these down and allow fall-through in the cases.